types/src/share.rs

```diff
@@ -45,14 +48,54 @@ pub struct Share {
 }
 
 impl Share {
+    /// Create a new Share with given namespace, and data, appropriately setting share sequence
+    /// metadata.
+    /// If sequence length is provided, first share is created, with sequence length field included
+    /// and sequence start indicator set. Otherwise continuation share is created.
+    /// Provided data must fit into the share type and is zero padded, if it's shorter.
+    pub fn new_v0(namespace: Namespace, sequence_length: Option<u32>, data: &[u8]) -> Result<Self> {
```

Comment:
I'm not so sure about the new api in `Share`, why is it needed for those changes?

Reply:
there are sparse shares and compact shares, and they should be differentiated by the namespace. Also constructing the shares by hand doesn't seem like a usecase for celestia users. You can check the ShareBuilders in celestia-app, but in general I'd stick to the Blob::split_to_shares api for now

Reply:
These were functions I wanted to use mostly for testing, but I thought others might find useful. Seems like for both cases `Blob` would probably suffice, so removing this as suggested